beginner_source/quickstart/dataquickstart_tutorial.py

```diff
@@ -0,0 +1,212 @@
+# Init
+# -----------------
+#
+# The init function is used for all the first time operations when our Dataset is loaded. In this case we use it to load our annotation labels to memory and then keep track of the directory of our image file. Note that different types of data can take different init inputs. You are not limited to just an annotations file, directory path and transforms, but for images this is a standard practice.
+# A sample csv annotations file may look as follows: ::
+#
+#     tshirt1.jpg, 0
+#     tshirt2.jpg, 0
+#     ......
+#     ankleboot999.jpg, 9
+# 
+# Example:
+# 
+
+def __init__(self, annotations_file, img_dir, transform=None):
+    self.img_labels = pd.read_csv(annotations_file)
+    self.img_dir = img_dir
+    self.transform = transform
+
+#################################################################
+# __len__
+# -----------------
+#
+# The __len__ function is needed to return the number of samples in our dataset. 
+# 
+# Example:
+
+def __len__(self):
+    return len(self.img_labels)
+
+#################################################################
+# __getitem__
+# -----------------
+#
+# The __getitem__ function is the most important function in the Datasets interface. It takes a tensor or an index as input and returns a loaded sample from your dataset at the given indices.
+# 
+# If provided a tensor as an index, we convert the tensor to a list first. We then load the file at the given index from our image directory, as well as the image label from our pandas annotations DataFrame. This image and label are then wrapped in a single sample dictionary which we can apply a transform on and return. Transforms will be discussed in more detail in the next section: `Transforms <transforms_tutorial.html>`_
+# 
+# Example:
+#
+
+def __getitem__(self, idx):
+    if torch.is_tensor(idx):
+        idx = idx.tolist()
+    img_path = os.path.join(self.root_dir,
+                            self.img_labels.iloc[idx, 0])
+    image = read_image(img_path)
+    label = self.img_labels.iloc[idx, 1:]
+    sample = {'image': image, 'label': label}
+    if self.transform:
+        sample = self.transform(sample)
+    return sample 
+
+#################################################################
+# Preparing your data for training with DataLoaders
+# -------------------------------------------------
+#
+# Now we have an organized mechanism for managing data which is great, but there is still a lot of manual work we would have to do to train a model with our Dataset. 
+# 
+# For example we would have to manually maintain the code for: 
+#
+# * Batching 
+# * Shuffling 
+# * Parallel batch distribution 
+# 
+# The PyTorch Dataloader ``torch.utils.data.DataLoader`` is an iterator that handles all of this complexity for us, enabling us to load a dataset and focus on training our model.
+
+dataloader = DataLoader(clothing, batch_size=4, shuffle=True, num_workers=0)
```

Comment:
We need a minimal example showing how to iterate a Dataloader. 
```
for sample in dataloader:
    X = sample['image']
    y = sample['label']

    # do something
```

Reply:
ping for visibility

Reply:
Added!